zip-0312.rst

```diff
@@ -1,7 +1,466 @@
+    list of commitments issued in Round 1 by each participant, where each element
+    in the list indicates a NonZeroScalar identifier j and two commitment
+    Element values (hiding_nonce_commitment_j, binding_nonce_commitment_j).
+    This list MUST be sorted in ascending order by identifier.
+  - group_public_key, public key corresponding to the group signing key,
+    an Element.
+  - msg, the message to be signed, a byte string.
+  - randomizer_point, an element in G.
+
+  Outputs:
+  - True if the signature share is valid, and False otherwise.
+
+  def verify_signature_share(identifier, PK_i, comm_i, sig_share_i, commitment_list,
+                             group_public_key, msg, randomizer_point):
+    # Compute the randomized group public key
+    randomized_group_public_key = group_public_key + randomizer_point
+
+    # Compute the binding factors
+    binding_factor_list = compute_binding_factors(commitment_list, msg, randomizer_point)
+    binding_factor = binding_factor_for_participant(binding_factor_list, identifier)
+
+    # Compute the group commitment
+    group_commitment = compute_group_commitment(commitment_list, binding_factor_list)
+
+    # Compute the commitment share
+    (hiding_nonce_commitment, binding_nonce_commitment) = comm_i
+    comm_share = hiding_nonce_commitment + G.ScalarMult(binding_nonce_commitment, binding_factor)
+
+    # Compute the challenge
+    challenge = compute_challenge(group_commitment, randomized_group_public_key, msg)
+
+    # Compute the interpolating value
+    participant_list = participants_from_commitment_list(commitment_list)
+    lambda_i = derive_interpolating_value(identifier, participant_list)
+
+    # Compute relation values
+    l = G.ScalarBaseMult(sig_share_i)
+    r = comm_share + G.ScalarMult(PK_i, challenge * lambda_i)
+
+    return l == r
+
+
+
+Ciphersuites
+------------
+
+FROST(Jubjub, BLAKE2b-512)
+'''''''''''''''''''''''''''''
+
+This ciphersuite uses Jubjub for the Group and BLAKE2b-512 for the Hash function `H`
+meant to produce signatures indistinguishable from RedJubjub Sapling Spend
+Authorization Signatures as specified in [#protocol-concretespendauthsig]_.
+
+- Group: Jubjub [#protocol-jubjub]_
+
+  - Order: 6554484396890773809930967563523245729705921265872317281365359162392183254199 (see [#protocol-jubjub]_)
+  - Identity: as defined in [#protocol-jubjub]_
+  - RandomScalar(): Implemented by returning a uniformly random Scalar in the range
+    \[0, `G.Order()` - 1\]. Refer to {{frost-randomscalar}} for implementation guidance.
+  - SerializeElement(P): Implemented as :math:`\mathsf{repr}_\mathbb{J}(P)` as defined in [#protocol-jubjub]_
+  - DeserializeElement(P): Implemented as :math:`\mathsf{abst}_\mathbb{J}(P)` as defined in [#protocol-jubjub]_,
+    failing if :math:`\bot` is returned. Additionally, this function validates that the resulting
+    element is not the group identity element, returning an error if the check fails.
```

Comment:
Assuming that there is no intended distinction between "failing" and "returning an error", use the same wording.

Reply:
Done in f72d8cd